docs/project-config.md

```diff
@@ -0,0 +1,279 @@
+  // ...
+  plugins: [
+    new AlexaCli({
+      files: {
+        'path.to.file.json': 'Hello World!'
+      }
+    })
+  ],
+});
+```
+
+
+## Staging
+
+Project staging allows you to `build` for and `deploy` to specific staging environments. A common use case is to have a `dev` stage for local development and a `prod` stage hosted in the cloud.
+
+Here is an example, how this could look like:
+
+```js
+const project = new ProjectConfig({
+  // ...
+
+  defaultStage: 'dev',
+  stages: {
+    dev: {
+      endpoint: '${JOVO_WEBHOOK_URL}',
+      // ...
+    },
+    prod: {
+      endpoint: process.env.ENDPOINT_PROD,
+      // ...
+    }
+  }
+});
+```
+
+In the example above, each stage has its own [`endpoint`](#endpoint). The `dev` stage uses the Jovo Webhook URL, and the `prod` stage a URL to a hosted service that is provided via environment variables in a `.env` file.
+
+You can add any stage name to the `stages` object:
+
+```js
+const project = new ProjectConfig({
+  // ...
+  stages: {
+    someStage: {
+      // ...
+    }
+  }
+});
+```
+
+Inside this stage, you can add any configuration that you can also add outside a stage, even `plugins`:
+
+```js
+const project = new ProjectConfig({
+  // ...
+  stages: {
+    someStage: {
+      // ...
+      plugins: [
+        new AlexaCli({ skillId: 'someSkillId' });
+      ]
+    }
+  }
+});
+```
+
+By default, only the elements outside the `stages` are used. If a specific stage is active, all content from this stage is merged into the generic config.
+
+The active stage is determined in the following order (most active first):
+
+* The stage is added as flag, e.g. `jovov4 build --stage someStage`
+* The default stage is set with `defaultStage: 'someStage'`
+* The stage is set in the environment variables with `NODE_ENV=someStage`
+
+
+## Hooks
+
+It's sometimes necessary to make small adjustments before running a Jovo CLI command. CLI Hooks provide a way, to do just that. For example, a hook could be used to retrieve data from an API or a CMS before running the `build` command.
+
+Hooks can be added like this:
+
+```js
+const project = new ProjectConfig({
+  // ...
+  hooks: {
+    // ...
+  },
+});
+```
+
+You can into any command, for example `before.build` or `after.build`:
+
+```js
+const project = new ProjectConfig({
+  // ...
+  hooks: {
+    'before.build': [
+      () => { /* Do something here */ }
+    ]
+  },
+});
+```
+
+A first example to test hooks might be to log something:
+
+```js
+const project = new ProjectConfig({
+  // ...
+  hooks: {
+    'before.build': [
+      () => console.log('Starting the build process now');
+    ]
+  },
+});
+```
+
+However, hooks usually require a few more lines of code. We recommend placing each function in a separate file in a `hooks` folder and then referencing it in the `jovo.project.js` file ([you can find an example of this hook here](https://github.com/rubenaeg/fetch-language-model-test/blob/master/hooks/fetchLanguageModel.hook.js)):
```

Comment:
Would it be useful to host the example on a `jovotech/` repository?

Reply:
I think we can keep it for now while we're still in Alpha, and then add example to the official CLI repo?

Reply:
Sounds good. I'll update the repo then to be public instead of private.